firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistributionTesterApiClient.java

```diff
@@ -225,12 +229,22 @@ private String getFingerprintHashForPackage(Context context) {
       hash = AndroidUtilsLight.getPackageCertificateHashBytes(context, context.getPackageName());
 
       if (hash == null) {
+        LogWrapper.getInstance()
+            .e(
+                TAG
+                    + "Could not get fingerprint hash for X-Android-Cert header. Package is not signed: "
```

Comment:
> Can this case only be triggered when the package is not signed?

Yes, according to the javadoc: http://google3/java/com/google/android/gmscore/integ/client/common/src/com/google/android/gms/common/util/AndroidUtilsLight.java;l=59;rcl=418549552

> Does the for X-Android-Cert header part of the log convey useful information for the dev?

I think so. Because if this is impactful, it will probably cause a failed request to our backend. So it will allow them to recognize the connection between this error and the failed request. The response from the server might mention the missing header for example.